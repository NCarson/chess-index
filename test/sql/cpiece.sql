

select pretty('p'::cpiece);
select pretty('n'::cpiece);
select pretty('b'::cpiece);
select pretty('r'::cpiece);
select pretty('q'::cpiece);
select pretty('k'::cpiece);
select pretty('P'::cpiece);
select pretty('N'::cpiece);
select pretty('B'::cpiece);
select pretty('R'::cpiece);
select pretty('Q'::cpiece);
select pretty('K'::cpiece);

select expected_or_fail_bool('p'::piece  = 'p'::piece, true);
select expected_or_fail_bool('p'::piece != 'p'::piece, false);
select expected_or_fail_bool('p'::piece != 'K'::piece, true);
select expected_or_fail_bool('p'::piece  = 'K'::piece, false);

select expected_or_fail_bool(cpiece_value('P'::cpiece) =  1, true);
select expected_or_fail_bool(cpiece_value('N'::cpiece) =  3, true);
select expected_or_fail_bool(cpiece_value('B'::cpiece) =  3, true);
select expected_or_fail_bool(cpiece_value('R'::cpiece) =  5, true);
select expected_or_fail_bool(cpiece_value('Q'::cpiece) =  9, true);
select expected_or_fail_bool(cpiece_value('K'::cpiece) =  0, true);
select expected_or_fail_bool(cpiece_value('p'::cpiece) = -1, true);
select expected_or_fail_bool(cpiece_value('n'::cpiece) = -3, true);
select expected_or_fail_bool(cpiece_value('b'::cpiece) = -3, true);
select expected_or_fail_bool(cpiece_value('r'::cpiece) = -5, true);
select expected_or_fail_bool(cpiece_value('q'::cpiece) = -9, true);
select expected_or_fail_bool(cpiece_value('k'::cpiece) =  0, true);
