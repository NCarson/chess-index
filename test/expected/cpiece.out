select pretty('p'::cpiece);
 pretty 
--------
 ♟
(1 row)

select pretty('n'::cpiece);
 pretty 
--------
 ♞
(1 row)

select pretty('b'::cpiece);
 pretty 
--------
 ♝
(1 row)

select pretty('r'::cpiece);
 pretty 
--------
 ♜
(1 row)

select pretty('q'::cpiece);
 pretty 
--------
 ♛
(1 row)

select pretty('k'::cpiece);
 pretty 
--------
 ♚
(1 row)

select pretty('P'::cpiece);
 pretty 
--------
 ♙
(1 row)

select pretty('N'::cpiece);
 pretty 
--------
 ♘
(1 row)

select pretty('B'::cpiece);
 pretty 
--------
 ♗
(1 row)

select pretty('R'::cpiece);
 pretty 
--------
 ♖
(1 row)

select pretty('Q'::cpiece);
 pretty 
--------
 ♕
(1 row)

select pretty('K'::cpiece);
 pretty 
--------
 ♔
(1 row)

select expected_or_fail_bool('p'::piece  = 'p'::piece, true);
 expected_or_fail_bool 
-----------------------
 
(1 row)

select expected_or_fail_bool('p'::piece != 'p'::piece, false);
 expected_or_fail_bool 
-----------------------
 
(1 row)

select expected_or_fail_bool('p'::piece != 'K'::piece, true);
 expected_or_fail_bool 
-----------------------
 
(1 row)

select expected_or_fail_bool('p'::piece  = 'K'::piece, false);
 expected_or_fail_bool 
-----------------------
 
(1 row)

select expected_or_fail_bool(cpiece_value('P'::cpiece) =  1, true);
 expected_or_fail_bool 
-----------------------
 
(1 row)

select expected_or_fail_bool(cpiece_value('N'::cpiece) =  3, true);
 expected_or_fail_bool 
-----------------------
 
(1 row)

select expected_or_fail_bool(cpiece_value('B'::cpiece) =  3, true);
 expected_or_fail_bool 
-----------------------
 
(1 row)

select expected_or_fail_bool(cpiece_value('R'::cpiece) =  5, true);
 expected_or_fail_bool 
-----------------------
 
(1 row)

select expected_or_fail_bool(cpiece_value('Q'::cpiece) =  9, true);
 expected_or_fail_bool 
-----------------------
 
(1 row)

select expected_or_fail_bool(cpiece_value('K'::cpiece) =  0, true);
 expected_or_fail_bool 
-----------------------
 
(1 row)

select expected_or_fail_bool(cpiece_value('p'::cpiece) = -1, true);
 expected_or_fail_bool 
-----------------------
 
(1 row)

select expected_or_fail_bool(cpiece_value('n'::cpiece) = -3, true);
 expected_or_fail_bool 
-----------------------
 
(1 row)

select expected_or_fail_bool(cpiece_value('b'::cpiece) = -3, true);
 expected_or_fail_bool 
-----------------------
 
(1 row)

select expected_or_fail_bool(cpiece_value('r'::cpiece) = -5, true);
 expected_or_fail_bool 
-----------------------
 
(1 row)

select expected_or_fail_bool(cpiece_value('q'::cpiece) = -9, true);
 expected_or_fail_bool 
-----------------------
 
(1 row)

select expected_or_fail_bool(cpiece_value('k'::cpiece) =  0, true);
 expected_or_fail_bool 
-----------------------
 
(1 row)

